File: 2026-01/2026-01-19/energyConsumption.js
function compareEnergy(caloriesBurned, wattHoursUsed) {
  const CALORIES_TO_JOULES_CONVERSION_RATE = 4184;
  const WATT_HOUR_TO_JOULES_CONVERSION_RATE = 3600;

  const workoutEnergyJoules = caloriesBurned * CALORIES_TO_JOULES_CONVERSION_RATE;
  const devicesEnergyJoules = wattHoursUsed * WATT_HOUR_TO_JOULES_CONVERSION_RATE;

  return workoutEnergyJoules > devicesEnergyJoules ? "Workout" 
    : workoutEnergyJoules < devicesEnergyJoules ? "Devices" 
    : "Equal";
}

// --- TEST SUITE ---

const testsText = String.raw`
1. compareEnergy(250, 50) should return "Workout".
2. compareEnergy(100, 200) should return "Devices".
3. compareEnergy(450, 523) should return "Equal".
4. compareEnergy(300, 75) should return "Workout".
5. compareEnergy(200, 250) should return "Devices".
6. compareEnergy(900, 1046) should return "Equal".
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+?)\.?$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("--------------------------");
  console.log("🧪Starting Verification...");
  console.log("--------------------------");

  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(test.output);

    const comparison = Array.isArray(testOutput)
      ? arraysEqual(functionCallOutput, testOutput)
      : functionCallOutput === testOutput;

    if (comparison) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    }
    else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${testOutput}\nGot: ${functionCallOutput}`);
      ++failCount;
    }
  })

  console.log("----------------------------",
    failCount
    ? `\n⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`
    : "\n🎉SUCCESS: All tests PASSED."
  );
}

function arraysEqual(a, b) {
  if (a.length !== b.length)
    return false;

  for (let i = 0; i < a.length; ++i) {
    if (Array.isArray(a[i])) {
      if (Array.isArray(b[i])) {
        if (!arraysEqual(a[i], b[i]))
          return false;
      }
      else 
        return false;
    }

    else if (a[i] !== b[i])
      return false;
  }

  return true;
}

runTests(testData);

File: 01 January/01-19/energy_consumption.py
def compare_energy(calories_burned, watt_hours_used):
    CALORIES_TO_JOULES_CONVERSION_RATE = 4184
    WATT_HOUR_TO_JOULES_CONVERSION_RATE = 3600

    workout_energy_joules = calories_burned * CALORIES_TO_JOULES_CONVERSION_RATE
    devices_energy_joules = watt_hours_used * WATT_HOUR_TO_JOULES_CONVERSION_RATE

    return "Workout" if workout_energy_joules > devices_energy_joules \
        else "Devices" if workout_energy_joules < devices_energy_joules \
        else "Equal"

# --- TEST SUITE ---

tests_text = r'''
1. compare_energy(250, 50) should return "Workout".
2. compare_energy(100, 200) should return "Devices".
3. compare_energy(450, 523) should return "Equal".
4. compare_energy(300, 75) should return "Workout".
5. compare_energy(200, 250) should return "Devices".
6. compare_energy(900, 1046) should return "Equal".
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+?)\.?$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("--------------------------")
    print("🧪 Starting Verification...")
    print("--------------------------")

    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            fail_count += 1

    print("----------------------------",
        f"\n⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED." if fail_count
        else "\n🎉 SUCCESS: All tests PASSED."
    )

run_tests(test_data)